src/http/api/views/account.follows.viewer.ts

```diff
@@ -0,0 +1,228 @@
+    id: number;
+    ap_id: string;
+    name: string;
+    username: string;
+    avatar_url: string;
+}
+
+export class AccountFollowsViewer {
+    /**
+     * @param db Database client
+     */
+    constructor(
+        private readonly db: Knex,
+        private readonly fedifyContextFactory: FedifyContextFactory,
+    ) {}
+
+    async getFollows(
+        type: string,
+        siteDefaultAccount: Account,
+        offset: number,
+    ): Promise<AccountFollowsView> {
+        if (!siteDefaultAccount.id) {
+            throw new Error('Site default account not found');
+        }
+
+        // Retrieve data
+        const getAccounts =
+            type === 'following'
+                ? this.getFollowingAccounts.bind(this)
+                : this.getFollowerAccounts.bind(this);
+        const getAccountsCount =
+            type === 'following'
+                ? this.getFollowingAccountsCount.bind(this)
+                : this.getFollowerAccountsCount.bind(this);
+
+        const results = await getAccounts(
+            siteDefaultAccount.id,
+            FOLLOWS_LIMIT,
+            offset,
+        );
+        const total = await getAccountsCount(siteDefaultAccount.id);
+
+        const next =
+            total > offset + FOLLOWS_LIMIT
+                ? (offset + FOLLOWS_LIMIT).toString()
+                : null;
+
+        const accounts: AccountInfo[] = [];
+
+        for (const result of results) {
+            accounts.push({
+                id: String(result.id),
+                name: result.name || '',
+                handle: getAccountHandle(
+                    new URL(result.ap_id).host,
+                    result.username,
+                ),
+                avatarUrl: result.avatar_url || '',
+                isFollowing:
+                    type === 'following'
+                        ? true
+                        : await this.checkIfAccountIsFollowing(
+                              siteDefaultAccount.id,
+                              result.id,
+                          ),
+            });
+        }
+
+        const accountFollowsView: AccountFollowsView = {
+            accounts: accounts,
+            total: total,
+            next: next,
+        };
+        return accountFollowsView;
+    }
+
+    /**
+     * Get the number of accounts that the provided account is following
+     *
+     * @param accountId id of the account
+     */
+    async getFollowingAccountsCount(accountId: number | null): Promise<number> {
+        if (!accountId) {
+            return 0;
+        }
+
+        const result = await this.db('follows')
+            .where('follower_id', accountId)
+            .count('*', { as: 'count' });
+
+        return Number(result[0].count);
+    }
+
+    /**
+     * Get the number of accounts that are following the provided account
+     *
+     * @param accountId id of the account
+     */
+    async getFollowerAccountsCount(accountId: number | null): Promise<number> {
```

Comment:
Do we need this param to be nullable? I think it's preferable to have it as `number` and do any checking in the method above (which afaict already happens)

This makes it easier to reason about and we don't have to deal with default return values (e.g. an account that doesn't exist doesn't have 0 followers, it has undefined/null followers - the question doesn't make sense)